### PR TITLE
Solidity Ultra Verifier with tests

### PR DESCRIPTION
**Pull request by LHerskind**
# Description

This PR works to move the verifier tests from https://github.com/aztecProtocol/aztec-verifier-contracts into barretenberg where the verifier key is generated.

The pr ONLY includes tests specific to the ULTRA verifier, as the standard verifier contract in aztec-verifier-contracts are using the linearization trick that have been deprecated and removed from barretenberg. 

We add `solidity_key_gen` and `solidity_proof_gen` executables that build keys and proofs for the verifier tests. These must be rebuild to properly test the verifier.

TODO: 
- [ ] Consider if `sol_gen` should be moved to solidity helpers instead of it current location in `plonk/proof_system/verification_key`.   
- [ ] Decide when to run the GHA, possibly with nix?

# Checklist:

- [ ] I have reviewed my diff in github, line by line.
- [ ] Every change is related to the PR description.
- [ ] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this pull request to the issue(s) that it resolves.
- [ ] There are no unexpected formatting changes, superfluous debug logs, or commented-out code.
- [ ] There are no circuit changes, OR specifications in `/markdown/specs` have been updated.
- [ ] There are no circuit changes, OR a cryptographer has been assigned for review.
- [ ] I've updated any terraform that needs updating (e.g. environment variables) for deployment.
- [ ] The branch has been rebased against the head of its merge target.
- [ ] I'm happy for the PR to be merged at the reviewer's next convenience.
- [ ] New functions, classes, etc. have been documented according to the doxygen comment format. Classes and structs must have `@brief` describing the intended functionality.
- [ ] If existing code has been modified, such documentation has been added or updated.

### CODE REVIEW

File: cpp/src/barretenberg/solidity_helpers/utils/utils.hpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#include <iostream>
+#include <bitset>
+
+// TODO Remove
```
Comment (Maddiaa0) [inline cpp/src/barretenberg/solidity_helpers/utils/utils.hpp]: removeable

File: cpp/src/barretenberg/solidity_helpers/circuits/recursive_circuit.hpp [hunk 2]
```diff
@@ -0,0 +1,144 @@
+#include "barretenberg/transcript/transcript.hpp"
+#include "barretenberg/plonk/proof_system/proving_key/serialize.hpp"
+#include "barretenberg/stdlib/primitives/curves/bn254.hpp"
+#include "barretenberg/stdlib/recursion/verifier/verifier.hpp"
+#include "barretenberg/stdlib/recursion/verifier/program_settings.hpp"
+#include "barretenberg/ecc/curves/bn254/fq12.hpp"
+#include "barretenberg/ecc/curves/bn254/pairing.hpp"
+
+//#include <transcript/transcript.hpp>
```
Comment (Maddiaa0) [inline cpp/src/barretenberg/solidity_helpers/circuits/recursive_circuit.hpp]: removable

File: cpp/src/barretenberg/solidity_helpers/circuits/blake_circuit.hpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
+#include "barretenberg/stdlib/primitives/field/field.hpp"
+#include "barretenberg/stdlib/primitives/witness/witness.hpp"
+#include "barretenberg/stdlib/hash/blake2s/blake2s.hpp"
+
+//#include <stdlib/primitives/field/field.hpp>
```
Comment (Maddiaa0) [inline cpp/src/barretenberg/solidity_helpers/circuits/blake_circuit.hpp]: removable

File: cpp/src/barretenberg/solidity_helpers/circuits/add_2_circuit.hpp [hunk 4]
```diff
@@ -0,0 +1,32 @@
+#include "barretenberg/stdlib/primitives/field/field.hpp"
+#include "barretenberg/stdlib/primitives/witness/witness.hpp"
+#include "barretenberg/stdlib/primitives/uint/uint.hpp"
+#include "barretenberg/stdlib/primitives/bool/bool.hpp"
+
+// #include <stdlib/primitives/field/field.hpp>
```
Comment (Maddiaa0) [inline cpp/src/barretenberg/solidity_helpers/circuits/add_2_circuit.hpp]: removable

**Review by Maddiaa0 — APPROVED** [review]

File: cpp/src/barretenberg/solidity_helpers/circuits/add_2_circuit.hpp (hunk 4 above)
Comment (LHerskind) [reply cpp/src/barretenberg/solidity_helpers/circuits/add_2_circuit.hpp]: Addressed in https://github.com/AztecProtocol/barretenberg/pull/363/commits/ca5f638df7a462a6b102d6a7e86a94759a95d335

File: cpp/src/barretenberg/solidity_helpers/circuits/blake_circuit.hpp (hunk 3 above)
Comment (LHerskind) [reply cpp/src/barretenberg/solidity_helpers/circuits/blake_circuit.hpp]: Addressed in https://github.com/AztecProtocol/barretenberg/pull/363/commits/ca5f638df7a462a6b102d6a7e86a94759a95d335

File: cpp/src/barretenberg/solidity_helpers/circuits/recursive_circuit.hpp (hunk 2 above)
Comment (LHerskind) [reply cpp/src/barretenberg/solidity_helpers/circuits/recursive_circuit.hpp]: Addressed in https://github.com/AztecProtocol/barretenberg/pull/363/commits/ca5f638df7a462a6b102d6a7e86a94759a95d335

File: cpp/src/barretenberg/solidity_helpers/utils/utils.hpp (hunk 1 above)
Comment (LHerskind) [reply cpp/src/barretenberg/solidity_helpers/utils/utils.hpp]: Addressed in https://github.com/AztecProtocol/barretenberg/pull/363/commits/ca5f638df7a462a6b102d6a7e86a94759a95d335

**Review by Maddiaa0 — APPROVED** [review]

**Pull request merged**